bin/runtime/src/migrations/custom_nomination_pools.rs

```diff
@@ -0,0 +1,139 @@
+pub use nomination_pools::CustomMigrateToV2;
+
+mod nomination_pools {
+    use codec::{Decode, Encode, Error, Input};
+    use frame_support::{
+        log,
+        traits::{OnRuntimeUpgrade, StorageVersion},
+    };
+    use pallet_nomination_pools::{
+        BalanceOf, BondedPools, Config, Metadata, Pallet, PoolId, PoolMember, PoolMembers,
+        ReversePoolIdLookup, RewardPool, RewardPools, SubPoolsStorage,
+    };
+    use sp_core::{Get, U256};
+    use sp_std::{collections::btree_set::BTreeSet, prelude::*};
+
+    use crate::sp_api_hidden_includes_construct_runtime::hidden_include::dispatch::GetStorageVersion; // sick
+    use crate::Weight;
+
+    #[derive(Decode)]
+    pub struct OldRewardPool<B> {
+        pub balance: B,
+        pub total_earnings: B,
+        pub points: U256,
+    }
+
+    enum EitherRewardPool<T: Config, B> {
+        Old(OldRewardPool<B>),
+        New(RewardPool<T>),
+    }
+
+    impl<T: Config, B: Decode> Decode for EitherRewardPool<T, B> {
+        fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
+            let len = input.remaining_len()?.unwrap_or_default();
+            let mut buffer = vec![0; len];
+            input.read(&mut buffer)?;
+            if let Ok(new) = OldRewardPool::<B>::decode(&mut buffer.clone().as_slice()) {
+                return Ok(EitherRewardPool::Old(new));
+            }
+
+            RewardPool::<T>::decode(&mut buffer.as_slice()).map(|old| EitherRewardPool::New(old))
+        }
+    }
+
+    fn dissolve_pool<T: Config>(id: PoolId) {
```

Comment:
We don't want to do as original `dissolve_pool` does? ie  that part " All accounts are dusted and the leftover
 of the reward account is returned to the depositor."

Now I think unfortunately there's more work needed, as original `dissolve_pool` is called in the situation the pool depositor withdraw as last one points. So we can't just remove those PoolMembers, we need to try at least to transfer their tokens back.

Reply:
done (as discussed offline)